gateway/src/shard/impl.rs

```diff
@@ -380,6 +380,35 @@ impl Shard {
         Ok(())
     }
 
+    /// Send a close code to Discord causing a resume or a reconnect.
+    ///
+    /// # Recommended close codes
+    ///
+    /// | Re-identify | Resume |
+    /// | ----------- | ------ |
+    /// |        1000 |   1012 |
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`CommandError::SessionInactive`] if the shard's
+    /// session is inactive.
+    ///
+    /// [`SessionInactiveError`]: struct.SessionInactiveError.html
+    pub fn send_close_code(&self, code: u16) -> Result<(), CommandError> {
```

Comment:
Why don't we just make a more useful method that can send any tungstenite message

Reply:
The main issue I see with it is that we then expose tungestenite such that we cannot update it without breaking changes.

Reply:
We already expose tungstenite in the API https://docs.rs/twilight-gateway/0.2.5/twilight_gateway/shard/struct.ShardSink.html#impl-Sink%3CMessage%3E-1

Reply:
I had forgotten about that, then I guess I see no issues with it.